feat: Add ignore_older option for local.file_match (#2245)
grafana/alloy: docs/sources/reference/components/local/local.file_match.md

The following arguments are supported:

Name                | Type                | Description                                                                                | Default | Required
---------------     | ------------------- | ------------------------------------------------------------------------------------------ |---------| --------
`path_targets`      | `list(map(string))` | Targets to expand; looks for glob patterns on the  `__path__` and `__path_exclude__` keys. |         | yes
`sync_period`       | `duration`          | How often to sync filesystem and targets.                                                  | `"10s"` | no

> `ignore_older_than` | `duration`          | Ignores files which are modified before this duration                                      |         | no

Reviewer: ```suggestion
`ignore_older_than` | `duration`          | Ignores files which are modified before this duration.                                     |         | no
```

The following arguments are supported:

Name                | Type                | Description                                                                                | Default | Required
---------------     | ------------------- | ------------------------------------------------------------------------------------------ |---------| --------
`path_targets`      | `list(map(string))` | Targets to expand; looks for glob patterns on the  `__path__` and `__path_exclude__` keys. |         | yes
`sync_period`       | `duration`          | How often to sync filesystem and targets.                                                  | `"10s"` | no
`ignore_older_than` | `duration`          | Ignores files which are modified before this duration.                                     |  `"0s"` | no

`path_targets` uses [doublestar][] style paths.
* `/tmp/**/*.log` will match all subfolders of `tmp` and include any files that end in `*.log`.
* `/tmp/apache/*.log` will match only files in `/tmp/apache/` that end in `*.log`.
* `/tmp/**` will match all subfolders of `tmp`, `tmp` itself, and all files.

> * When `ignore_older_than` is set to "0s" the component will not ignore any files

Reviewer: ```suggestion
* `local.file_match` doesn't ignore files when `ignore_older_than` is set to the default, `0s`.
```

> * `local.file_match` doesn't ignore files when `ignore_older_than` is set to the default, `0s`.

Reviewer 2: ```suggestion

`local.file_match` doesn't ignore files when `ignore_older_than` is set to the default, `0s`.
```

--------------- | ------------------- | ------------------------------------------------------------------------------------------ |---------| --------
`path_targets`  | `list(map(string))` | Targets to expand; looks for glob patterns on the  `__path__` and `__path_exclude__` keys. |         | yes
`sync_period`   | `duration`          | How often to sync filesystem and targets.                                                  | `"10s"` | no

> `ignore_older`  | `duration`          | Ignores files which are modified before this duration                                      |         | no

Author: @wildum If we add that as "0s" wouldn't that be misleading that we ignore files older than current timestamp while the actual implementation considers all the files ?

> `ignore_older`  | `duration`          | Ignores files which are modified before this duration                                      |         | no

Reviewer 2: Let's ask the doc expert, @clayton-cornell would you rather:

- have the default value explicitly documented as "0s" with a line under saying that 0 means that it will consider all the files. Although this is more transparent, this might be misleading for users to see the "0s".
- not document the "0s" to implicitly say that this functionality is deactivated by default

or something else?

> `ignore_older`  | `duration`          | Ignores files which are modified before this duration                                      |         | no

Author: Yup made the change Please take a look
